### PR TITLE
[Remove Stream/Optional usage] LPS-177345 Remove_Stream_usages_for_module_segments_test

### PR DESCRIPTION
**Pull request by jeffwu0724**
Hi Tina, Dante,

This is for ticket https://issues.liferay.com/browse/LPS-177345.
Thanks for checking!

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:tango

**Comment by Kvale1733** [comment]
@jeffwu0724 No regressions found.

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/criteria/contributor/test/UserSegmentsCriteriaContributorTest.java [hunk 1]
```diff
@@ -194,15 +195,19 @@ public void testGetFieldsWithComplexEntity() throws Exception {
 		List<Field> fields = segmentsCriteriaContributor.getFields(
 			_getMockPortletRequest());
 
-		Stream<Field> stream = fields.stream();
+		Set<String> complexEntityFieldNames = new HashSet<>(
+			TransformUtil.transform(
```
Comment (dantewang) [inline modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/criteria/contributor/test/UserSegmentsCriteriaContributorTest.java:198]: @jeffwu0724 Please just use a for-loop to populate the set.

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/criteria/contributor/test/UserSegmentsCriteriaContributorTest.java [hunk 2]
```diff
@@ -225,47 +230,36 @@ public void testGetFieldsWithOptions() throws Exception {
...
+		List<Field> filteredFields = ListUtil.filter(
+			segmentsCriteriaContributor.getFields(_getMockPortletRequest()),
 			field -> StringUtil.endsWith(
 				field.getName(),
-				Normalizer.normalizeIdentifier(expandoColumn.getName()))
-		).findFirst();
+				Normalizer.normalizeIdentifier(expandoColumn.getName())));
+
+		Optional<Field> fieldOptional = Optional.ofNullable(
+			filteredFields.get(0));
```
Comment (dantewang) [inline modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/criteria/contributor/test/UserSegmentsCriteriaContributorTest.java]: @jeffwu0724 Although it's a test, we don't want to always go through the whole list when we can shortcut earlier. Please make this and the change below a for-loop with shortcut.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Pull request closed without merging**